src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,144 @@
+            ((uint8_t)(value)[2] << 8) | (uint8_t)(value)[3])
+
+namespace mixxx {
+
+namespace {
+
+const mixxx::Logger kLogger("StemInfoImporter");
+constexpr int kSupportedStemVersion = 1;
+const QStringList kStemFileExtensions = {".stem.mp4", ".stem.m4a"};
+
+const uint32_t kAtomHeaderSize = 8; // 4 bytes (unsigned integer) + 4 char
+const uint32_t kStemManifestAtomPath[] = {
+        ATOM_TYPE("moov"),
+        ATOM_TYPE("udta"),
+        ATOM_TYPE("stem"),
+        0 // This indicate end of the path
+};
+
+/// @brief Seek the reader the atom requested.
+/// @param reader The IODevice to search MP4 atom in
+/// @param path The list of atom to traverse in the tree, from top to bottom.
+/// Must be null terminated
+/// @param boxSize The size of the box currently under the cursor to focus the
+/// search in
+/// @param pathIdx The level of the tree currently search (index of path)
+/// @return the size of the box data if found, -1 otherwise
+uint32_t seekTillAtom(QIODevice& reader,
+        const uint32_t path[],
+        uint32_t boxSize = 0,
+        uint32_t pathIdx = 0) {
+    if (!path[pathIdx]) {
+        return boxSize;
+    }
+
+    uint32_t byteRead = 0;
+    char buffer[kAtomHeaderSize];
+    while (!reader.atEnd() || (boxSize && byteRead >= boxSize)) {
+        reader.read(buffer, kAtomHeaderSize);
+        byteRead += boxSize;
+        if (ATOM_TYPE(buffer + 4) == path[pathIdx]) {
+            return seekTillAtom(reader, path, ATOM_TYPE(buffer) - kAtomHeaderSize, pathIdx + 1);
+        }
+        reader.skip(ATOM_TYPE(buffer) - kAtomHeaderSize);
+    }
+    return -1;
+}
+
+} // anonymous namespace
+
+// static
+bool StemInfoImporter::isStemFile(
+        const QString& fileName) {
+    for (const QString& ext : kStemFileExtensions) {
+        if (fileName.endsWith(ext)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+QList<StemInfo> StemInfoImporter::importStemInfos(
+        const QString& filePath) {
+    // Fetch the STEM manifest which contain stream details
+    auto file = QFile(filePath);
+    if (!file.open(QIODeviceBase::ReadOnly)) {
+        kLogger.warning()
+                << "Failed to open input file"
+                << filePath;
+        return {};
+    }
+
+    uint32_t manifestSize;
+    if (!(manifestSize = seekTillAtom(file, kStemManifestAtomPath))) {
+        kLogger.warning()
+                << "Failed to find the steam manifest in the file"
+                << filePath;
+        return {};
+    }
+
+    auto jsonData = QJsonDocument::fromJson(file.read(manifestSize));
+    VERIFY_OR_DEBUG_ASSERT(jsonData.isObject()) {
+        kLogger.warning()
+                << "Failed to extract the manifest data"
+                << filePath;
+        return {};
+    };
+
+    auto json = jsonData.object();
+
+    // Check the manifest version
+    auto version = json.value("version").toInteger(-1);
+    if (version <= 0) {
```

Comment:
I'd rather simplify all this by checking for `version==1` and bailing out on anything else.

Reply:
I'm confused. In other comments, we aim to make the implementation be permissive and allow flexibility. 
Happy to change the debug at `108` to warning if we want it to be visible with the default log level, but it goes against the other efforts to hard fail here for no reason.

Reply:
this is mostly theoretical as there was no update to the this standard since released nearly 10 years ago, but since this is a single number, an increment likely means a breaking change. so I think it would be cleaner to outright reject something that we have no guarantee of parsing correctly

Reply:
> I think it would be cleaner to outright reject something that we have no guarantee of parsing correctly

That statement can be applied on many of the other requested changes. Again, I think it's best to keep and trying best effort on continuing loading. There are more tests coming later to validate the data (channel layout, matching sample rate, ...) so if this is indeed a breaking change, then it will abort later one, when it gets indeed stuck. At that stage, it's best to carry on IMO. 
As a middle ground, I will update the subsequent `debug` log to a `warning` so it gets displayed by default.